### ISSUE TITLE
Build failed on *BSD and Mac OS

### ISSUE THREAD

**Issue by Roman3349** [logimic/shape#23]
```
~/shape/deploy/Unix_Makefiles ~/shape
~/shape
~/shape/build/Unix_Makefiles ~/shape
CMake Warning (dev) at CMakeLists.txt:1 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    CMAKE_PROJECT_VERSION
    CMAKE_PROJECT_VERSION_MAJOR
    CMAKE_PROJECT_VERSION_MINOR
    CMAKE_PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.

-- CMAKE_CXX_STANDARD:     
-- CMAKE_CXX_FLAGS:          -fPIC -std=c++11 -pthread -Wall -Wextra -Werror=format-security -fstack-protector-strong
-- CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
-- CMAKE_CXX_FLAGS_DEBUG:   -g
-- CMAKE_C_FLAGS:            -Wall -Wextra -Werror=format-security -fstack-protector-strong
-- CMAKE_C_FLAGS_RELEASE:   -O3 -DNDEBUG
-- CMAKE_C_FLAGS_DEBUG:     -g
-- DeclareShapeComponent(): shape::TraceFormatService
-- Component shape::TraceFormatService configured to /homes/eva/xo/xondra58/shape/build/Unix_Makefiles/TraceFormatService/shape__TraceFormatService.hxx
-- DeclareShapeComponent(): shape::TraceFileService
-- Component shape::TraceFileService configured to /homes/eva/xo/xondra58/shape/build/Unix_Makefiles/TraceFileService/shape__TraceFileService.hxx
-- DeclareShapeComponent(): shapeExpl::ComponentTemplate
-- Component shapeExpl::ComponentTemplate configured to /homes/eva/xo/xondra58/shape/build/Unix_Makefiles/ComponentTemplate/shapeExpl__ComponentTemplate.hxx
-- DeclareShapeComponent(): shape::Example1_Thread
-- Component shape::Example1_Thread configured to /homes/eva/xo/xondra58/shape/build/Unix_Makefiles/examples/Example1_Thread/shape__Example1_Thread.hxx
-- GoogleTest_INCLUDE_DIR: /homes/eva/xo/xondra58/shape/external/googletest/googletest/include;/homes/eva/xo/xondra58/shape/external/googletest/googlemock/include
-- GoogleTest_LIBRARIES: optimized;gtest;optimized;gmock;debug;gtestd;debug;gmockd
-- Configuring done
-- Generating done
-- Build files have been written to: /homes/eva/xo/xondra58/shape/build/Unix_Makefiles
~/shape
Scanning dependencies of target launcher
[  3%] Building CXX object launcher/CMakeFiles/launcher.dir/Component.cpp.o
In file included from /homes/eva/xo/xondra58/shape/launcher/Component.cpp:19:
In file included from /homes/eva/xo/xondra58/shape/launcher/ComponentManager.h:21:
In file included from /homes/eva/xo/xondra58/shape/launcher/ConfigurationManager.h:21:
In file included from /homes/eva/xo/xondra58/shape/include/IConfiguration.h:19:
/homes/eva/xo/xondra58/shape/include/ShapePropertiesJson.h:95:10: warning: 'isSubsetOf' overrides a member function but is not marked
      'override' [-Winconsistent-missing-override]
    bool isSubsetOf(const Properties & props) const
         ^
/homes/eva/xo/xondra58/shape/include/ShapeProperties.h:34:18: note: overridden virtual function is here
    virtual bool isSubsetOf(const Properties & props) const = 0;
                 ^
In file included from /homes/eva/xo/xondra58/shape/launcher/Component.cpp:17:
In file included from /homes/eva/xo/xondra58/shape/include/StaticComponentMap.h:19:
In file included from /homes/eva/xo/xondra58/shape/include/ShapeComponent.h:20:
In file included from /homes/eva/xo/xondra58/shape/include/ShapeProperties.h:4:
/homes/eva/xo/xondra58/shape/include/rapidjson/document.h:1854:106: error: no member named 'Set' in
      'rapidjson::internal::TypeHelper<rapidjson::GenericValue<rapidjson::UTF8<char>,
      rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >, std::__1::__bit_const_reference<std::__1::vector<bool,
      std::__1::allocator<bool> > > >'
    ValueType& Set(const T& data, AllocatorType& allocator) { return internal::TypeHelper<ValueType, T>::Set(*this, data, allocator); }
                                                                                                         ^
/homes/eva/xo/xondra58/shape/include/ShapePropertiesJson.h:469:15: note: in instantiation of function template specialization
      'rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator>
      >::Set<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >' requested here
          ins.Set(it, alloc);
              ^
/homes/eva/xo/xondra58/shape/include/ShapePropertiesJson.h:271:14: note: in instantiation of function template specialization
      'shape::PropertiesJson::setMemberAsVector<bool>' requested here
      return setMemberAsVector(name, val);
             ^
1 warning and 1 error generated.
gmake[2]: *** [launcher/CMakeFiles/launcher.dir/build.make:63: launcher/CMakeFiles/launcher.dir/Component.cpp.o] Chyba 1
gmake[1]: *** [CMakeFiles/Makefile2:197: launcher/CMakeFiles/launcher.dir/all] Chyba 2
gmake: *** [Makefile:130: all] Chyba 2
[  3%] Building CXX object launcher/CMakeFiles/launcher.dir/Component.cpp.o
In file included from /homes/eva/xo/xondra58/shape/launcher/Component.cpp:19:
In file included from /homes/eva/xo/xondra58/shape/launcher/ComponentManager.h:21:
In file included from /homes/eva/xo/xondra58/shape/launcher/ConfigurationManager.h:21:
In file included from /homes/eva/xo/xondra58/shape/include/IConfiguration.h:19:
/homes/eva/xo/xondra58/shape/include/ShapePropertiesJson.h:95:10: warning: 'isSubsetOf' overrides a member function but is not marked
      'override' [-Winconsistent-missing-override]
    bool isSubsetOf(const Properties & props) const
         ^
/homes/eva/xo/xondra58/shape/include/ShapeProperties.h:34:18: note: overridden virtual function is here
    virtual bool isSubsetOf(const Properties & props) const = 0;
                 ^
In file included from /homes/eva/xo/xondra58/shape/launcher/Component.cpp:17:
In file included from /homes/eva/xo/xondra58/shape/include/StaticComponentMap.h:19:
In file included from /homes/eva/xo/xondra58/shape/include/ShapeComponent.h:20:
In file included from /homes/eva/xo/xondra58/shape/include/ShapeProperties.h:4:
/homes/eva/xo/xondra58/shape/include/rapidjson/document.h:1854:106: error: no member named 'Set' in
      'rapidjson::internal::TypeHelper<rapidjson::GenericValue<rapidjson::UTF8<char>,
      rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >, std::__1::__bit_const_reference<std::__1::vector<bool,
      std::__1::allocator<bool> > > >'
    ValueType& Set(const T& data, AllocatorType& allocator) { return internal::TypeHelper<ValueType, T>::Set(*this, data, allocator); }
                                                                                                         ^
/homes/eva/xo/xondra58/shape/include/ShapePropertiesJson.h:469:15: note: in instantiation of function template specialization
      'rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator>
      >::Set<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >' requested here
          ins.Set(it, alloc);
              ^
/homes/eva/xo/xondra58/shape/include/ShapePropertiesJson.h:271:14: note: in instantiation of function template specialization
      'shape::PropertiesJson::setMemberAsVector<bool>' requested here
      return setMemberAsVector(name, val);
             ^
1 warning and 1 error generated.
gmake[2]: *** [launcher/CMakeFiles/launcher.dir/build.make:63: launcher/CMakeFiles/launcher.dir/Component.cpp.o] Chyba 1
gmake[1]: *** [CMakeFiles/Makefile2:197: launcher/CMakeFiles/launcher.dir/all] Chyba 2
gmake: *** [Makefile:130: all] Chyba 2
```

**Comment by fmikulu** [comment]
@Roman3349 please describe how to reproduce it or make a pull request if you have a solution.

**Comment by Roman3349** [comment]
@fmikulu Unfortunately I have not a solution but I can you provide access to FreeBSD machine.

**How to reproduce**
 - create virtual machine (e.g. in VirtualBox) with some BSD UNIX OS (e.g. FreeBSD)
 - install dependencies
 - build shape

**Comment by fmikulu** [comment]
Fix is in d779c27fbe4fbd4b31973745ef7cee187bb9f48e
